### PR TITLE
Implement Jaccard Index shortcut for metrics

### LINKED ISSUE #1680: Adding Jaccard index for metrics
## 🚀 Feature

I propose implementing Jaccard index (https://en.wikipedia.org/wiki/Jaccard_index) for ignite metrics. I believe it should be doable given we already support confusion matrices.

### PR DESCRIPTION
**Pull request by Moh-Yakoub**
Fixes #1680 

Description:

Adding a Jaccard Index shortcut. https://en.wikipedia.org/wiki/Jaccard_index

It does the same computation as the IoU metric. Adding this shortcut as this is a more common name for the metric that IoU.

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/confusion_matrix.py [hunk 1]
```diff
@@ -323,3 +323,31 @@ def ignore_index_fn(dice_vector: torch.Tensor) -> torch.Tensor:
...
+        MetricsLambda
+
+    Examples:
+
+    .. code-block:: python
+
+        train_evaluator = ...
+
+        cm = ConfusionMatrix(num_classes=num_classes)
+        JaccardIndex(cm, ignore_index=0).attach(train_evaluator, 'IoU')
```
Comment (vfdev-5) [inline ignite/metrics/confusion_matrix.py:347]:
```suggestion
        JaccardIndex(cm, ignore_index=0).attach(train_evaluator, 'JaccardIndex')
```

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the PR @Moh-Yakoub !
I minor thing to fix in the docs

File: ignite/metrics/confusion_matrix.py [hunk 2]
```diff
@@ -323,3 +323,31 @@ def ignore_index_fn(dice_vector: torch.Tensor) -> torch.Tensor:
         return MetricsLambda(ignore_index_fn, dice)
     else:
         return dice
+
+
+def JaccardIndex(cm: ConfusionMatrix, ignore_index: Optional[int] = None) -> MetricsLambda:
+    r"""Calculates the Jaccard Index using :class:`~ignite.metrics.ConfusionMatrix` metric.
+    This is the same like the IoU metric
+    .. math:: \text{J}(A, B) = \frac{ \lvert A \cap B \rvert }{ \lvert A \cup B \rvert }
```
Comment (vfdev-5) [inline ignite/metrics/confusion_matrix.py:331]: This is badly rendered: https://deploy-preview-1682--pytorch-ignite-preview.netlify.app/metrics.html#ignite.metrics.JaccardIndex

File: ignite/metrics/confusion_matrix.py [hunk 3]
```diff
@@ -323,3 +323,31 @@ def ignore_index_fn(dice_vector: torch.Tensor) -> torch.Tensor:
         return MetricsLambda(ignore_index_fn, dice)
     else:
         return dice
+
+
+def JaccardIndex(cm: ConfusionMatrix, ignore_index: Optional[int] = None) -> MetricsLambda:
+    r"""Calculates the Jaccard Index using :class:`~ignite.metrics.ConfusionMatrix` metric.
+    This is the same like the IoU metric
```
Comment (vfdev-5) [inline ignite/metrics/confusion_matrix.py:330]:
```suggestion
    Implementation is based on :meth:`~ignite.metrics.IoU`.
```

**Comment by vfdev-5** [comment]
Thanks for the updates @Moh-Yakoub , few more things to update in the docs.

File: ignite/metrics/confusion_matrix.py (hunk 2 above)
Comment (Moh-Yakoub) [reply ignite/metrics/confusion_matrix.py:332]: Screen Shot 2021-02-22 at 11 26 19 PM

Seems to be fixed now!

**Comment by vfdev-5** [comment]
@Moh-Yakoub I removed the type annotations from docstring as it is ongoing work now, see https://github.com/pytorch/ignite/issues/1665

**Review by vfdev-5 — APPROVED** [review]
LGTM!

**Pull request merged**